pipeline/src/transformers/addressables/exhibitionHighlightTour.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  asText,
+  asTitle,
+  isFilledLinkToDocumentWithData,
+  isNotUndefined,
+} from '@weco/content-pipeline/src/helpers/type-guards';
+import { ExhibitionHighlightTourPrismicDocument } from '@weco/content-pipeline/src/types/prismic';
+import { ElasticsearchAddressableExhibitionHighlightTour } from '@weco/content-pipeline/src/types/transformed';
+
+export const transformAddressableExhibitionHighlightTour = (
+  document: ExhibitionHighlightTourPrismicDocument
+): ElasticsearchAddressableExhibitionHighlightTour[] => {
+  const { data, id, uid: documentUid } = document;
+  const relatedExhibition = isFilledLinkToDocumentWithData(
+    data.related_exhibition
+  )
+    ? data.related_exhibition
+    : undefined;
+  const exhibitionTitleField = relatedExhibition
+    ? relatedExhibition.data.title
+    : undefined;
+  const exhibitionTitle = exhibitionTitleField && asTitle(exhibitionTitleField);
+  const introText = data.intro_text && asText(data.intro_text);
+  const primaryImage = relatedExhibition
+    ? relatedExhibition.data.promo?.[0]?.primary
+    : undefined;
+
+  const promoCaption = primaryImage?.caption && asText(primaryImage.caption);
+
+  const description = introText ?? promoCaption;
+
+  const audioTitle = `${exhibitionTitle} audio highlight tour`;
+  const bslTitle = `${exhibitionTitle} British Sign Language tour`;
+
+  function getBody(type: 'audio' | 'bsl') {
+    return data.slices
+      ?.map(s => {
+        const subtitles = s.primary.subtitles?.length
+          ? s.primary.subtitles
+          : undefined;
+        const transcript = s.primary.transcript?.length
+          ? s.primary.transcript
+          : undefined;
+        const texts =
+          type === 'audio'
+            ? (transcript ?? subtitles)
+            : (subtitles ?? transcript);
+        return [s.primary.title?.map(t => t.text), texts?.map(t => t.text)]
+          .filter(isNotUndefined)
+          .flat();
+      })
+      .flat();
+  }
+
+  const audioBody = getBody('audio');
+  const bslBody = getBody('bsl');
+
+  const uid = documentUid || undefined;
+
+  const shared = {
+    id,
+    uid,
+    display: {
+      type: 'Exhibition highlight tour' as const,
+      id,
+      uid,
+      description,
+    },
+    query: {
+      type: 'Exhibition highlight tour' as const,
+      description,
```

Comment:
Could query contain both intro text and promo caption?

Reply:
I didn't dig into it, but maybe because of the spreads below

Reply:
Good point – no, it wasn't supposed to. Will update.

Reply:
[In Stories I put both the standfirst and the promo caption](https://github.com/wellcomecollection/content-api/pull/170/files#diff-d95d969e0baadd0614d2ff56d1784de3b351649b2331ec34a3f1eed9ae403981R59), so I thought maybe it could be a similar thing here, but maybe it's too repetitive..? 🤔